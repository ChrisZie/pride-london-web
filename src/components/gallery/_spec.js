import React from 'react'
import { shallow } from 'enzyme'
import Gallery from '.'

describe('Image', () => {
  const images = [
    {
      name: 'Test 1',
      description: 'Test 1 Description',
      originalAlt: 'test-image',
      thumbnailAlt: 'test-image',
      original: `foo`,
      thumbnail: `foo`,
    },
    {
      name: 'Test 2',
      description: null,
      originalAlt: 'test-2-image',
      thumbnailAlt: 'test-2-image',
      original: `foo`,
      thumbnail: `foo`,
    },
  ]

  it('renders when given props', () => {
    const wrapper = shallow(<Gallery images={images} />)
    expect(wrapper).toMatchSnapshot()
  })
})
