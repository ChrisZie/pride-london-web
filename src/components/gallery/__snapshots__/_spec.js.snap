// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders when given props 1`] = `
<Fragment>
  <ImageGallery
    additionalClass="custom-gallery"
    autoPlay={false}
    disableArrowKeys={false}
    disableSwipe={false}
    disableThumbnailScroll={false}
    flickThreshold={0.4}
    indexSeparator=" / "
    infinite={true}
    isRTL={false}
    items={
      Array [
        Object {
          "description": "Test 1 Description",
          "name": "Test 1",
          "original": "foo",
          "originalAlt": "test-image",
          "thumbnail": "foo",
          "thumbnailAlt": "test-image",
        },
        Object {
          "description": null,
          "name": "Test 2",
          "original": "foo",
          "originalAlt": "test-2-image",
          "thumbnail": "foo",
          "thumbnailAlt": "test-2-image",
        },
      ]
    }
    lazyLoad={true}
    onSlide={[Function]}
    preventDefaultTouchmoveEvent={false}
    renderFullscreenButton={[Function]}
    renderLeftNav={[Function]}
    renderPlayPauseButton={[Function]}
    renderRightNav={[Function]}
    showBullets={false}
    showFullscreenButton={true}
    showIndex={false}
    showNav={true}
    showPlayButton={false}
    showThumbnails={true}
    slideDuration={450}
    slideInterval={3000}
    startIndex={0}
    stopPropagation={false}
    swipeThreshold={30}
    swipingTransitionDuration={0}
    thumbnailPosition="top"
    useBrowserFullscreen={true}
    useTranslate3D={true}
  />
  <p>
    Test 1 Description
  </p>
</Fragment>
`;
